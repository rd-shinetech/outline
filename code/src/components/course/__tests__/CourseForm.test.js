import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {CourseForm} from '../CourseForm';



function setup(heading) {
    const props = {
        handleSubmit: jest.fn(),
        pristine: true,
        reset: jest.fn(),
        submitting: false,
        heading: heading,
        authors: [],
        handleSave: jest.fn(),
        handleCancel: jest.fn(),
    };

    return shallow(<CourseForm {...props}/>);
}


describe('CourseForm.test.js', () => {

    it('renders without crashing', () => {
        const wrapper = setup('Add');
        expect(wrapper.length).toEqual(1);
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });


    it('renders form and display "Add" in h1', () => {
        const wrapper = setup('Add');
        expect(wrapper.find('h1').text()).toEqual('Add');
    });


    it('displays "Edit" in h1', () => {
        const wrapper = setup('Edit');
        expect(wrapper.find('h1').text()).toEqual('Edit');
    });

    
});

