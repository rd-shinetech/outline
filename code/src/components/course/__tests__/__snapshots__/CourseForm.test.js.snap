exports[`CourseForm.test.js renders without crashing 1`] = `
<form>
  <h1>
    Add
  </h1>
  <Field
    component={[Function]}
    label="Title"
    name="title"
    placeholder="Title of the course"
    type="text" />
  <Field
    component={[Function]}
    label="Author"
    name="authorId"
    options={Array []} />
  <Field
    component={[Function]}
    label="Category"
    name="category"
    placeholder="Category of the course"
    type="text" />
  <Field
    component={[Function]}
    label="Length"
    name="length"
    placeholder="Lenght of course in minutes or hours"
    type="text" />
  <div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit">
      <i
        aria-hidden="true"
        className="fa fa-paper-plane-o" />
       Submit
    </button>
    <button
      className="btn btn-default btn-space"
      disabled={true}
      onClick={[Function]}
      type="button">
      Clear Values
    </button>
    <button
      className="btn btn-default btn-space"
      onClick={[Function]}
      type="button">
      Cancel
    </button>
  </div>
</form>
`;
